openedx/core/djangoapps/credentials/signals.py

```diff
@@ -83,7 +84,12 @@ def send_grade_if_interesting(user, course_run_key, mode, status, letter_grade,
             )
         return
 
-    # Grab mode/status if we don't have them in hand
+    # Grab mode/status from cert call
+    if course_cert_data:
+        key = (user.id, str(course_run_key))
```

Comment:
I believe you should move this logic back into the management command and pass it in already defined arguments `mode` and `status` of this function.